### PR TITLE
Added network watcher APIs and tests

### PR DESCRIPTION
**Pull request by irrogozh**
## Description

Network SDK regenerated from a new swagger spec with added network watcher APIs.
Link to a pull request for swagger: https://github.com/Azure/azure-rest-api-specs/pull/923
AutoRest version: 1.0.0-Nightly20170202
Link on swagger spec: https://github.com/Azure/azure-rest-api-specs-pr/blob/master/arm-network/compositeNetworkClient.json or
https://github.com/irrogozh/azure-rest-api-specs-pr/blob/master/arm-network/compositeNetworkClient.json

I didn't update the link in generate.cmd because I don't know which one I should use, from Azure private repo or from my own public one.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@irrogozh__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (irrogozh). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Review by twitchax — APPROVED** [review]
LGTM - @irrogozh, can you fill out the PR checklist, please? :)

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

File: src/ResourceManagement/Network/Network.sln [hunk 1]
```diff
@@ -3,8 +3,12 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
 VisualStudioVersion = 14.0.25420.1
 MinimumVisualStudioVersion = 10.0.40219.1
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.Azure.Management.Compute", "..\Compute\Microsoft.Azure.Management.Compute\Microsoft.Azure.Management.Compute.xproj", "{CBE97730-45F5-448E-88E9-55DF94D65B77}"
+EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.Azure.Management.Network", "Microsoft.Azure.Management.Network\Microsoft.Azure.Management.Network.xproj", "{4F22B186-DEB4-4B95-BA58-0187F0E32993}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.Azure.Management.Storage", "..\Storage\Microsoft.Azure.Management.Storage\Microsoft.Azure.Management.Storage.xproj", "{A9B44E1B-C26F-44AB-B13F-F22742371125}"
+EndProject
```
Comment (cormacpayne) [inline src/ResourceManagement/Network/Network.sln]: @irrogozh why were Compute and Storage added to this file? It looks like you are trying to use them for tests, but you already have a reference to their published packages in the `Network.Tests/project.json`, so you don't need to reference the path to their code in this file.

**Review by cormacpayne — CHANGES_REQUESTED** [review]
@irrogozh Hey Irina, in addition to the comment I left in the code

- you will need to update the version in [project.json](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Network/Microsoft.Azure.Management.Network/project.json#L2) to whatever the version should be for release

- you will need to update the version in [AssemblyInfo.cs](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Network/Microsoft.Azure.Management.Network/Properties/AssemblyInfo.cs#L11) with the same version

File: src/ResourceManagement/Network/Network.sln (hunk 1 above)
Comment (irrogozh) [inline src/ResourceManagement/Network/Network.sln]: Fixed

**Comment by DeepakRajendranMsft** [comment]
@markcowl @cormacpayne can this be merged now please. 

Thanks,
deepak.

**Review by cormacpayne — APPROVED** [review]

**Pull request merged**